### ISSUE TITLE
tickValues no longer works with ordinal scales

### ISSUE THREAD

**Issue by johanrd** [d3/d3#2029]
If you run http://tributary.io/inlet/4966973 with the latest d3, the labels are not rendered. Error message: «Invalid value» (NaN).

![skjermbilde-2014-09-22-kl -09 24 22b](https://cloud.githubusercontent.com/assets/4601554/4352842/480e6fb4-422b-11e4-8c4a-4d4ad73aa4f8.jpg)

**Comment by mizzao** [comment]
@jasondavies I just ran into this issue, and the problem is not with rangeBands or rangePoints.

A few versions ago (at least 3.1.4) d3 would allow ordinal scales with different ticks displayed using `tickValues` with a `d3.svg.axis`. This is no longer possible; if you try and do that, d3 attempts to compute the scale position using the **tick label** instead of the **scale value**, resulting in an NaN because this label is not defined on the scale. This would seem to be a bug.

Sometimes it's desirable to have different scale fields versus tick fields, for example if the scale is for an encoded userId and the label is the username, and the data uses the userId, but you don't want to display an unreadable label.

To summarize, you can reproduce with
- Define an `d3.scale.ordinal` with `.domain(array)`
- Define a `d3.svg.axis` using the above scale and `.tickValues(labelArray)`
- Try to call the axis on an SVG element. You will get the above error.

You will not get the error without defining `tickValues`, the original elements of `array` will happily show up as the labels. If you define `tickValues`, you will get the screenshot above where a single label is drawn but not positioned properly because the error is thrown.

**Comment by mizzao** [comment]
@johanrd it would be good to change the title to something like "tickValues no longer works with ordinal scales".

**Comment by mizzao** [comment]
@jasondavies, can we reopen this issue? It turns out that this doesn't work with log scales as well, i.e. you can't use `tickValues` on a `d3.svg.axis` that is using a `d3.scale.log` - it is just ignored.

**Comment by jasondavies** [comment]
I don’t think it’s a bug: [axis.tickValues](https://github.com/mbostock/d3/wiki/SVG-Axes#tickValues) sets the tick *values* to be used.  In other words, you are defining values that are in the scale’s domain to be used as ticks.  In other words, values for which we show a tick mark and possibly a label.

If you want to change the tick formatting, e.g. to generate human-friendly labels, then you should use [axis.tickFormat](https://github.com/mbostock/d3/wiki/SVG-Axes#tickFormat).

**Comment by mizzao** [comment]
I see - `axis.tickValues` must come from the domain and not generate the format themselves. Understood.
